emc/tooldata/tooldata_common.cc:253:40 strncat truncates

A reminder.

```
emc/tooldata/tooldata_common.cc: In function ‘void tooldata_format_toolline(int, bool, CANON_TOOL_TABLE, char*)’:
emc/tooldata/tooldata_common.cc:253:40: warning: ‘char* __builtin_strncat(char*, const char*, long unsigned int)’ output may be truncated copying between 230 and 246 bytes from a string of length 254 [-Wstringop-truncation]
  253 |                                 strncat(formatted_line,tmp,space); \
      |                                        ^
emc/tooldata/tooldata_common.cc:262:5: note: in expansion of macro ‘F_ITEM’
  262 |     F_ITEM(diameter,       "D");
      |     ^~~~~~
emc/tooldata/tooldata_common.cc:253:40: warning: ‘char* __builtin_strncat(char*, const char*, long unsigned int)’ output may be truncated copying between 230 and 246 bytes from a string of length 254 [-Wstringop-truncation]
  253 |                                 strncat(formatted_line,tmp,space); \
      |                                        ^
emc/tooldata/tooldata_common.cc:262:5: note: in expansion of macro ‘F_ITEM’
  262 |     F_ITEM(diameter,       "D");
```

--- a/src/emc/tooldata/tooldata_common.cc
+++ b/src/emc/tooldata/tooldata_common.cc
@@ -245,12 +245,13 @@ void tooldata_format_toolline (int idx,
     space -= len;
 // format zero float values as %.0f for brevity
 #define F_ITEM(item,letter) if (!ignore_zero_values || tdata.item) { \
+                                char local_tmp[64] = {}; \
                                 if (tdata.item) { \
-                                    len = snprintf(tmp,sizeof(tmp)," " letter "%+f", tdata.item); \
+                                    len = snprintf(local_tmp,sizeof(local_tmp)," " letter "%+f", tdata.item); \
                                 } else { \
-                                    len = snprintf(tmp,sizeof(tmp)," " letter "%.0f",tdata.item); \
+                                    len = snprintf(local_tmp,sizeof(local_tmp)," " letter "%.0f",tdata.item); \
                                 } \
-                                strncat(formatted_line,tmp,space); \
+                                strncat(formatted_line,local_tmp,space); \
                                 space -= len; \
                             }
 #define I_ITEM(item,letter) if (!ignore_zero_values || tdata.item) { \

